.github/workflows/build.yaml

```diff
@@ -0,0 +1,64 @@
+name: Build and Deploy
+
+on:
+  push:
+    branches:
+      - main
+      - develop
+      - 'release/*'
+      - feature/add_github_actions
+
+jobs:  
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Get Revision
+        run: |
+          VERSION_TAG=$(./gradlew -q printVersion)
+          echo "VERSION_TAG=$VERSION_TAG" >> $GITHUB_ENV
+          IMAGE_TAGS="${VERSION_TAG}.$(date '+%Y.%m.%d')"
+          echo "IMAGE_SEMANTIC_HASH=$IMAGE_TAGS" >> $GITHUB_ENV
+          if [[ "${{ github.ref }}" == "refs/heads/main" ]]; then IMAGE_TAGS="$VERSION_TAG,latest"; fi;
+          echo "IMAGE_TAGS=$IMAGE_TAGS" >> $GITHUB_ENV
+      - name: GitGuardian scan
+        uses: GitGuardian/ggshield/actions/secret@v1.23.0
+        env:
+          GITHUB_PUSH_BEFORE_SHA: ${{ github.event.before }}
+          GITHUB_PUSH_BASE_SHA: ${{ github.event.base }}
+          GITHUB_PULL_BASE_SHA: ${{ github.event.pull_request.base.sha }}
+          GITHUB_DEFAULT_BRANCH: ${{ github.event.repository.default_branch }}
+          GITGUARDIAN_API_KEY: ${{ secrets.GITGUARDIAN_API_KEY }}
+    
+      - name: Build with Gradle
+        uses: eskatos/gradle-command-action@v1
+        with:
+          arguments: 'bootJar :test --info --build-cache'
+
+      - name: Login to Azure Container Registry
+        uses: azure/docker-login@v1
+        with:
+          login-server: oaftech.azurecr.io
+          username: ${{ vars.ACR_PULL_USERNAME }}
+          password: ${{ secrets.ACR_PULL_PASSWORD }}
+
+      - name: Build Docker image
+        run: docker build . -t oaftech.azurecr.io/oneacrefund/paygops-ams-connector:${{ env.IMAGE_TAGS }}
+
+      - name: Push image to registry
+        uses: docker/login-action@v3.0.0
+        with:
+          registry: oaftech.azurecr.io
+          username: ${{ vars.ACR_PULL_USERNAME }}
+          password: ${{ secrets.ACR_PULL_PASSWORD }}
+      - run: |
+          docker push oaftech.azurecr.io/oneacrefund/paygops-ams-connector:${{ env.IMAGE_TAGS }}  
```

Comment:
If we're also going to run this for PRs, are we okay with pushing an image to the registry each time a commit is made in the source branch? The existing azure pipeline only pushes based on certain conditions. I don't see the equivalent of those conditions here.